tests/v2/validation/workloads/workload_test.go

```diff
@@ -192,6 +193,91 @@ func (w *WorkloadTestSuite) TestWorkloadStatefulset() {
+
+	containerName := namegen.AppendRandomString("updatetestcontainer")
+	newContainerTemplate := workloads.NewContainer(containerName,
+		"redis",
+		corev1.PullAlways,
+		[]corev1.VolumeMount{},
+		[]corev1.EnvFromSource{},
+		nil,
+		nil,
+		nil,
+	)
+	upgradeDeployment.Spec.Template.Spec.Containers = []corev1.Container{newContainerTemplate}
+
+	upgradeDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, upgradeDeployment)
+	require.NoError(w.T(), err)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"1", "2"}, "redis", 2)
+
+	containerName = namegen.AppendRandomString("updatetestcontainertwo")
+	newContainerTemplate = workloads.NewContainer(containerName,
+		"ubuntu",
+		corev1.PullAlways,
+		[]corev1.VolumeMount{},
+		[]corev1.EnvFromSource{},
+		nil,
+		nil,
+		nil,
+	)
+	newContainerTemplate.TTY = true
+	newContainerTemplate.Stdin = true
+	upgradeDeployment.Spec.Template.Spec.Containers = []corev1.Container{newContainerTemplate}
+
+	_, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, upgradeDeployment)
+	require.NoError(w.T(), err)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"1", "2", "3"}, "ubuntu", 2)
+
+	logRollback, err := deployment.RollbackDeployment(w.client, w.cluster.ID, namespace.Name, upgradeDeployment, 1)
+	require.NoError(w.T(), err)
+	require.NotEmpty(w.T(), logRollback)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"2", "3", "4"}, "nginx", 2)
+
+	logRollback, err = deployment.RollbackDeployment(w.client, w.cluster.ID, namespace.Name, upgradeDeployment, 2)
+	require.NoError(w.T(), err)
+	require.NotEmpty(w.T(), logRollback)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"3", "4", "5"}, "redis", 2)
+
+	logRollback, err = deployment.RollbackDeployment(w.client, w.cluster.ID, namespace.Name, upgradeDeployment, 3)
+	require.NoError(w.T(), err)
+	require.NotEmpty(w.T(), logRollback)
+
+	validateRolloutHistory(w, namespace.Name, upgradeDeployment, []string{"4", "5", "6"}, "ubuntu", 2)
+}
+
+func validateRolloutHistory(w *WorkloadTestSuite, namespaceName string, appv1Deployment *appv1.Deployment, expectedRevisions []string, image string, expectedContainerCount int) {
+	err := charts.WatchAndWaitDeployments(w.client, w.cluster.ID, namespaceName, metav1.ListOptions{
+		FieldSelector: "metadata.name=" + appv1Deployment.Name,
+	})
+	require.NoError(w.T(), err)
+
+	err = pods.WatchAndWaitPodContainerRunning(w.client, w.cluster.ID, namespaceName, appv1Deployment)
+	require.NoError(w.T(), err)
+
+	revisions, err := deployment.RolloutHistoryDeployment(w.client, w.cluster.ID, namespaceName, appv1Deployment)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), expectedRevisions, revisions)
+
+	countPods, err := pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespaceName, image)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), expectedContainerCount, countPods)
```

Comment:
since this is not rollout history validation, can we move this to the test itself or change the function name to something else? its a bit misleading with validateRolloutHistory

Reply:
Yes, I can